engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/hashing/UpdateByStateManager.java

```diff
@@ -8,33 +8,53 @@
 import org.apache.commons.lang3.mutable.MutableInt;
 import org.jetbrains.annotations.NotNull;
 
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
 /**
  * The basis for operators that participate in an updateBy operation.
  */
-public interface ChunkedUpdateByStateManager {
-    void add(final SafeCloseable bc,
+public abstract class UpdateByStateManager {
```

Comment:
why class and not interface?

Looking at the actual hash tables there is a fair bit of duplication in the TypedBases.  I wonder if we should be collapsing them together; or if that is more trouble than it is worth.  The duplication of just the key sources seems like not really worth having a class instead of something more swappable.

Reply:
Re: why class and not interface, having a constructor is useful.

I think the `TypedBase`s should be collapsed.  `Incremental` contains all of `AddOnly` plus probing so should be easy to collapse these into one.